### PR TITLE
Add new rules to Checkstyle - final touch

### PR DESCRIPTION
**Pull request by wkurniawan07**
The capstone for Checkstyle rules.

- Modified some of the existing rules:
  - Method name: relaxed the rule so that things like `testAbc_expectsThis_returnsThat` are allowed.
  - Redundant modifier (Sun's config): added a few tokens. Few violations fixed.

- Added a bunch of new rules:
  - Upper "L" (Google's config): `0L` instead of `0l`. One violation.
  - Outer type file name (Google's config): outer class `ABC` must be in `ABC.java`. No violations.
  - One top level class (Google's config): `ABC.java` must only have one outer class `ABC` (inner classes not counted). No violations.
  - Uncommented main: self-explanatory. Suppressed for client scripts for obvious reason. One violation in `TeamEvalResultTest`, but it looks useful so I commented it out for now instead of removing.
  - Catch parameter name: allows `e`, `t`, or any camel-case word of length two or more, no numbers allowed. Slightly more relaxed than the suggested rule which also prohibits length of two. Many violations.
  - Abbreviation as word in name (Google's config): `getURL` must be `getUrl`. Many violations.
  - Constant name: the usual rule of all caps and underscore. Relaxed the rule to also allow `*Logic`, `*Db`, `logic`, `dataBundle` which are often used either in tests or in logic/db layer, and `log` which is the logger. Few violations. Suppressed for client scripts because most fields there are constants.

- Suppressions in Checkstyle now have to be specific, like in PMD/ESLint. The syntax is `CHECKSTYLE.OFF:RuleNameOne|RuleNameTwo` (separated by `|`), similarly for the `ON` counterpart. This is reflected in our static analysis documentation.

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/InstructorPrivileges.java [hunk 1]
```diff
@@ -12,97 +12,97 @@
...
+        PRIVILEGES_CUSTOM.put(Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_COMMENT_IN_SECTIONS, false);
+        PRIVILEGES_CUSTOM.put(Const.ParamsNames.INSTRUCTOR_PERMISSION_GIVE_COMMENT_IN_SECTIONS, false);
+        PRIVILEGES_CUSTOM.put(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_COMMENT_IN_SECTIONS, false);
+        PRIVILEGES_CUSTOM.put(Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_SESSION_IN_SECTIONS, false);
+        PRIVILEGES_CUSTOM.put(Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS, false);
+        PRIVILEGES_CUSTOM.put(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_SESSION_COMMENT_IN_SECTIONS, false);
     }
 
-    private static final String[] courseLevelOnlyList = new String[] {
+    private static final String[] LIST_COURSE_LEVEL_ONLY = new String[] {
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/datatransfer/InstructorPrivileges.java]: This (and similar ones) will be changed to `*_LEVEL_ONLY_LIST` to be consistent.

**Comment by wkurniawan07** [comment]
@JosephineKwa tagging you if you're interested 😛

File: src/test/java/teammates/test/cases/logic/TeamEvalResultTest.java [hunk 2]
```diff
@@ -491,7 +491,7 @@ private void verifyNormalized(double[] expected, double[] input) {
                 Arrays.toString(TeamEvalResult.normalizeValues(input)));
     }
     
-    private static void showCalculationSteps(int[][] input) {
+    /*private static void showCalculationSteps(int[][] input) {
```
Comment (kanghj) [inline src/test/java/teammates/test/cases/logic/TeamEvalResultTest.java]: Hmm, if we are just commenting this out, then maybe let's just suppress the rule here?

Comment (wkurniawan07) [inline src/test/java/teammates/test/cases/logic/TeamEvalResultTest.java]: I could alternatively take it out and put it to a new class in the `client/` folder. It'll automatically be ignored by the uncommented main rule, and also from test coverage, by the virtue of it being in the `client/` folder. Although it's not a script to connect with the datastore. What do you think?

Comment (kanghj) [inline src/test/java/teammates/test/cases/logic/TeamEvalResultTest.java]: Hmm, we could try that. This test file doesn't seem like the right place for it anyway.

**Pull request merged**